src/System.Text.Json/src/System/Text/Json/CustomUncheckedBitArray.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json
+{
+    // Inspired by BitArray
+    // https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/BitArray.cs
+    internal struct CustomUncheckedBitArray
+    {
+        private int[] _array;
+
+        public int Length { get; private set; }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public CustomUncheckedBitArray(int bitLength, int integerLength)
```

Comment:
You're taking both the bitlength and the integerLength to avoid needing to bit shift the bitLength?  The only call site for this constructor uses constants... is the JIT not able to do constant propagation when the ctor gets inlined?